sdk/python/jobs/pipelines/1j_pipeline_with_pipeline_component/nyc_taxi_data_regression_with_pipeline_component/nyc_taxi_data_regression_with_pipeline_component.ipynb

```diff
@@ -0,0 +1,383 @@
+   "metadata": {},
+   "source": [
+    "## 1.2 Configure credential\n",
+    "\n",
+    "We are using `DefaultAzureCredential` to get access to workspace. \n",
+    "`DefaultAzureCredential` should be capable of handling most Azure SDK authentication scenarios. \n",
+    "\n",
+    "Reference for more available credentials if it does not work for you: [configure credential example](../../configuration.ipynb), [azure-identity reference doc](https://docs.microsoft.com/en-us/python/api/azure-identity/azure.identity?view=azure-python)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "try:\n",
+    "    credential = DefaultAzureCredential()\n",
+    "    # Check if given credential can get token successfully.\n",
+    "    credential.get_token(\"https://management.azure.com/.default\")\n",
+    "except Exception as ex:\n",
+    "    # Fall back to InteractiveBrowserCredential in case DefaultAzureCredential not work\n",
+    "    credential = InteractiveBrowserCredential()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 1.3 Get a handle to the workspace\n",
+    "\n",
+    "We use config file to connect to a workspace. The Azure ML workspace should be configured with computer cluster. [Check this notebook for configure a workspace](../../configuration.ipynb)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Get a handle to workspace\n",
+    "ml_client = MLClient.from_config(credential=credential)\n",
+    "\n",
+    "# Retrieve an already attached Azure Machine Learning Compute.\n",
+    "cluster_name = \"cpu-cluster\"\n",
+    "print(ml_client.compute.get(cluster_name))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# 2. Define and create components into workspace\n",
+    "## 2.1 Define pipeline component with data prep and transformation\n",
+    "### 2.1.1 Load components from YAML"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "components_dir = \"./data_pipeline\"\n",
+    "prep = load_component(source=f\"{components_dir}/prep_src/prep.yml\")\n",
+    "transform = load_component(source=f\"{components_dir}/transform_src/transform.yml\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 2.1.2 Define pipeline component with components"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# add meta info when invoke decorator.\n",
+    "@pipeline(display_name='data_pipeline',\n",
+    "    tags={'authoring':'sdk'},\n",
+    "    # version='123',\n",
+    "    description='pipeline component with data prep and transformation defined via yaml-meta.',\n",
+    "    is_deterministic=True\n",
```

Comment:
no is_deterministic for pipeline component?

Reply:
Good question, theoretically, pipeline component shouldn't have this, as it is more like container, this property may not work on it.